### PR TITLE
Enable and fix @typescript-eslint/semi

### PR DESCRIPTION
**Pull request by paulius-valiunas**

### CODE REVIEW

**Review by pmconne — APPROVED** [review]

**Review by bsteinbk — APPROVED** [review]

**Review by saskliutas — APPROVED** [review]

**Review by grigasp — APPROVED** [review]

**Review by calebmshafer — APPROVED** [review]

File: common/changes/@bentley/eslint-plugin/eslint-semi_2020-11-10-14-15.json [hunk 1]
```diff
@@ -0,0 +1,11 @@
+{
+  "changes": [
+    {
+      "packageName": "@bentley/eslint-plugin",
+      "comment": "",
```
Comment (calebmshafer) [inline common/changes/@bentley/eslint-plugin/eslint-semi_2020-11-10-14-15.json:5]: @paulius-valiunas can you put a comment in here about the change in defaults for imodeljs-recommended?  Consumers of the plugin are going to hit this as well.

Comment (paulius-valiunas) [reply common/changes/@bentley/eslint-plugin/eslint-semi_2020-11-10-14-15.json]: is this better?

**Comment by calebmshafer** [comment]
@EarlinLutz or @mgooding mind taking a look?

**Review by EarlinLutz — APPROVED** [review]
I bequeath Caleb an unlimited number of semicolons.

**Pull request merged**